src/v/kafka/server/write_at_offset_stm.cc

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "kafka/server/write_at_offset_stm.h"
+
+#include "cluster/snapshot.h"
+#include "kafka/server/logger.h"
+#include "model/batch_utils.h"
+
+#include <seastar/coroutine/as_future.hh>
+
+#include <algorithm>
+
+namespace kafka {
+
+namespace {
+struct local_snapshot
+  : serde::
+      envelope<local_snapshot, serde::version<0>, serde::compat_version<0>> {
+    auto serde_fields() { return std::tie(last_offset); }
+
+    kafka::offset last_offset;
```

Comment:
do we need to store this in the snapshot? we can initialize from the log?

Reply:
not really as we need to refer to committed data (committed in Raft sense)